hare/flows_test.go

```diff
@@ -404,6 +404,7 @@ func Test_multipleCPs(t *testing.T) {
 	var pubsubs []*pubsub.PubSub
 	scMap := NewSharedClock(totalNodes, totalCp, time.Duration(50*int(totalCp)*totalNodes)*time.Millisecond)
 	outputs := make([]map[types.LayerID]LayerOutput, totalNodes)
```

Comment:
thanks for submitting a PR @piersy to fix this. highly appreciated.
tho i don't think this fixes the data race. the race happens because the goroutines are modifying `output` at the same time. 
making sure the goroutines are done before we check the result is good, but not enough to fix the race.

Reply:
Good point, I wonder why this doesn't seem to be triggering any data races?

I'll update to sync over the map.